docs/02-developers/05-smart-contracts/05-foundry/verify-smart-contracts.md

```diff
@@ -0,0 +1,607 @@
+# Verify Smart Contracts Using Foundry Script on Rootstock
+
+---
+**Location**: `docs/02-developers/05-smart-contracts/05-foundry/verify-smart-contracts.md`
+**Sidebar Position**: 106
+**Tags**: [guides, developers, smart contracts, rsk, rootstock, foundry, verification, explorer]
+---
+
+## Overview
+
+Smart contract verification on Rootstock using Foundry provides developers with a streamlined, automated approach to make deployed contracts transparent and trustworthy. This comprehensive guide covers both command-line verification and automated script-based verification using the Rootstock Explorer API, ensuring your contracts are properly verified for community trust and interaction.
+
+## Prerequisites
+
+Before proceeding with verification, ensure you have completed the foundational setup:
+
+- **Foundry Installation & Configuration**: Follow the complete setup guide at [Configure Foundry for Rootstock](/developers/smart-contracts/foundry/configure-foundry-rootstock/)
+- **Deployed Smart Contract**: Your contract must be successfully deployed on Rootstock. See [Deploy Smart Contracts](/developers/smart-contracts/foundry/deploy-smart-contracts/) for deployment instructions
+- **Contract Source Code**: Access to the exact source code used for deployment
+- **Constructor Arguments**: Documentation of all constructor parameters used during deployment
+- **Environment Setup**: Properly configured `.env` file with necessary API keys and RPC URLs
+
+> **Important**: This guide assumes familiarity with Foundry basics. For foundational knowledge, review the [Foundry Project Creation](/developers/smart-contracts/foundry/create-foundry-project/) and [Smart Contract Development](/developers/smart-contracts/foundry/smart-contracts/) guides.
```

Comment:
Use the admonition components for TIP or Info: https://dev.rootstock.io/components/#admonitions